homeassistant/util/dt.py

```diff
@@ -76,7 +76,7 @@ def as_timestamp(dt_value):
     else:
         parsed_dt = parse_datetime(str(dt_value))
         if not parsed_dt:
-            raise ValueError("not a valid date/time.")
+            return None
```

Comment:
Although maybe exposed as a template function, this is our date utils and they should fail with an exception when incorrect information is passed in.